tt_eager/tensor/tensor_impl.hpp

```diff
@@ -560,43 +560,59 @@ inline Tensor pad(
                 &output_tensor_shape,
                 &input_tensor_start,
                 &pad_value_](const auto& input_buffer) {
-        // Check if input tensor fits in output tensor given the input tensor start indices
-        TT_ASSERT(input_tensor_shape[0] + input_tensor_start[0] <= output_tensor_shape[0]);
-        TT_ASSERT(input_tensor_shape[1] + input_tensor_start[1] <= output_tensor_shape[1]);
-        TT_ASSERT(input_tensor_shape[2] + input_tensor_start[2] <= output_tensor_shape[2]);
-        TT_ASSERT(input_tensor_shape[3] + input_tensor_start[3] <= output_tensor_shape[3]);
-
-        // Figure out pad size on each dim
-        uint32_t pad_size[4][2] = {
-            {input_tensor_start[0], output_tensor_shape[0] - input_tensor_shape[0] - input_tensor_start[0]},
-            {input_tensor_start[1], output_tensor_shape[1] - input_tensor_shape[1] - input_tensor_start[1]},
-            {input_tensor_start[2], output_tensor_shape[2] - input_tensor_shape[2] - input_tensor_start[2]},
-            {input_tensor_start[3], output_tensor_shape[3] - input_tensor_shape[3] - input_tensor_start[3]}};
-
-        const std::array<uint32_t, 4> output_tensor_strides = {
-            output_tensor_shape[1] * output_tensor_shape[2] * output_tensor_shape[3],
-            output_tensor_shape[2] * output_tensor_shape[3],
-            output_tensor_shape[3],
-            1};
+
+
+        auto compute_stride = [](const Shape& shape, uint32_t index) {
+            uint32_t stride = 1;
+            for (auto i = index + 1; i < shape.rank(); i++) {
+                stride *= shape[i];
+            }
+            return stride;
+        };
+
+        std::array<std::array<uint32_t, 2>, 4> pad_size{};
+
+        std::array<uint32_t, 4> input_tensor_strides{};
+        std::array<uint32_t, 4> output_tensor_strides{};
+
+        for (auto index = 0; index < output_tensor_shape.rank(); index++) {
+            // Check if input tensor fits in output tensor given the input tensor start indices
+            TT_ASSERT(
+                input_tensor_shape[index] + input_tensor_start[index] <= output_tensor_shape[index],
+                "Input tensor is out of bounds");
+
+            // Figure out pad size on each dim
+            pad_size[index + 4 - output_tensor_shape.rank()] = {
+                input_tensor_start[index],
+                output_tensor_shape[index] - input_tensor_shape[index] - input_tensor_start[index]};
+
+            input_tensor_strides[index + 4 - output_tensor_shape.rank()] = compute_stride(input_tensor_shape, index);
+            output_tensor_strides[index + 4 - output_tensor_shape.rank()] = compute_stride(output_tensor_shape, index);
+        }
+
+        auto N = input_tensor_shape.rank() >= 4 ? input_tensor_shape[-4] : 1;
+        auto C = input_tensor_shape.rank() >= 3 ? input_tensor_shape[-3] : 1;
+        auto H = input_tensor_shape[-2];
+        auto W = input_tensor_shape[-1];
```

Comment:
Is this working correctly for 5d, 6d, 7d, and 8d tensors?

Reply:
I updated the code further. It will now work with literally any rank